### ISSUE TITLE
Unhandled Exception: FileSystemException: Cannot retrieve length of file

### ISSUE THREAD

**Issue by Tizoner** [humblerookie/optimized_cached_image#7]
```
E/flutter (18091): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: FileSystemException: Cannot retrieve length of file, path = '/data/user/0/com.example.music_taming/cache/libCachedImageData/5ee35fd0-9142-11ea-ed86-8791e4bc3e42_tmp.jpg' (OS Error: No such file or directory, errno = 2)
[38;5;244mE/flutter (18091): #0      _File.throwIfError  (dart:io/file_impl.dart:645:7)[39;49m
[38;5;244mE/flutter (18091): #1      _File.lengthSync  (dart:io/file_impl.dart:387:5)[39;49m
[38;5;248mE/flutter (18091): #2      DefaultImageTransformer._scaleImageFile[39;49m
E/flutter (18091): <asynchronous suspension>
[38;5;248mE/flutter (18091): #3      DefaultImageTransformer.transform[39;49m
[38;5;248mE/flutter (18091): #4      ImageCacheManager.getFileStream.<anonymous closure>[39;49m
[38;5;244mE/flutter (18091): #5      _rootRunUnary  (dart:async/zone.dart:1192:38)[39;49m
[38;5;244mE/flutter (18091): #6      _CustomZone.runUnary  (dart:async/zone.dart:1085:19)[39;49m
[38;5;244mE/flutter (18091): #7      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:987:7)[39;49m
[38;5;244mE/flutter (18091): #8      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:339:11)[39;49m
[38;5;244mE/flutter (18091): #9      _DelayedData.perform  (dart:async/stream_impl.dart:594:14)[39;49m
[38;5;244mE/flutter (18091): #10     _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:710:11)[39;49m
[38;5;244mE/flutter (18091): #11     _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:670:7)[39;49m
[38;5;244mE/flutter (18091): #12     _rootRun  (dart:async/zone.dart:1180:38)[39;49m
[38;5;244mE/flutter (18091): #13     _CustomZone.run  (dart:async/zone.dart:1077:19)[39;49m
[38;5;244mE/flutter (18091): #14     _CustomZone.runGuarded  (dart:async/zone.dart:979:7)[39;49m
[38;5;244mE/flutter (18091): #15     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1019:23)[39;49m
[38;5;244mE/flutter (18091): #16     _rootRun  (dart:async/zone.dart:1184:13)[39;49m
[38;5;244mE/flutter (18091): #17     _CustomZone.run  (dart:async/zone.dart:1077:19)[39;49m
[38;5;244mE/flutter (18091): #18     _CustomZone.runGuarded  (dart:async/zone.dart:979:7)[39;49m
[38;5;244mE/flutter (18091): #19     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1019:23)[39;49m
[38;5;244mE/flutter (18091): #20     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)[39;49m
[38;5;244mE/flutter (18091): #21     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)[39;49m
```

Debug mode pointed to resizedFile.lengthSync() in _scaleImageFile
```
Future<FileInfo> _scaleImageFile(FileInfo info, int width, int height) async {
  File file = info.file;
  if (file.existsSync()) {
    String extension = p.extension(file.path) ?? '';
    final format = _compressionFormats[extension] ?? CompressFormat.png;
    final tmpFile = getTempFile(file, format);
    final srcSize = file.lengthSync();
    final screen = window.physicalSize;
    File resizedFile = await FlutterImageCompress.compressAndGetFile(
        file.path, tmpFile.path,
        minWidth: width ?? screen.width.toInt(),
        minHeight: height ?? screen.height.toInt(),
        format: format);
    if (resizedFile.lengthSync() < srcSize) {
      resizedFile.renameSync(file.path);
    } else {
      resizedFile.deleteSync();
    }
  }
  return info;
}
```

**Comment by Tizoner** [comment]
Though after I've cleaned cache and tried again error didn't appear.
Instead  it constantly print in console this line:
```
W/System  (19534): A resource failed to call close.
```

**Comment by humblerookie** [comment]
Could you try using this latest commit, this should resolve the file not found error.
```
optimized_cached_image:
    git:
      url: git@github.com:humblerookie/optimized_cached_image.git
      ref: 3cb501d
```
For some reason layout builder passes 0 , 0 as the constraints at times, this causes the compression step to fail. 

Regarding this
> W/System  (19534): A resource failed to call close.

This library is using` flutter_image_compress` and it turns out to be holding on connections/not closing them. Will raise an issue there

**Comment by Tizoner** [comment]
The problem here not only layout builder but also that it tries to delete files that may not exist.
```
E/flutter (14419): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: FileSystemException: Cannot delete file, path = '/data/user/0/com.example.music_taming/cache/libCachedImageData/3221a430-9178-11ea-eb7d-098233681d14_tmp.jpg' (OS Error: No such file or directory, errno = 2)
E/flutter (14419): #0      _File.throwIfError  (dart:io/file_impl.dart:645:7)
E/flutter (14419): #1      _File._deleteSync  (dart:io/file_impl.dart:308:5)
E/flutter (14419): #2      FileSystemEntity.deleteSync  (dart:io/file_system_entity.dart:460:47)
E/flutter (14419): #3      DefaultImageTransformer._scaleImageFile 
package:optimized_cached_image/image_transformer.dart:70
E/flutter (14419): <asynchronous suspension>
E/flutter (14419): #4      DefaultImageTransformer.transform 
E/flutter (14419): #5      ImageCacheManager.getFileStream.<anonymous closure> 
package:optimized_cached_image/image_cache_manager.dart:81
E/flutter (14419): #6      _rootRunUnary  (dart:async/zone.dart:1192:38)
E/flutter (14419): #7      _CustomZone.runUnary  (dart:async/zone.dart:1085:19)
E/flutter (14419): #8      _CustomZone.runUnaryGuarded  (dart:async/zone.dart:987:7)
E/flutter (14419): #9      _BufferingStreamSubscription._sendData  (dart:async/stream_impl.dart:339:11)
E/flutter (14419): #10     _DelayedData.perform  (dart:async/stream_impl.dart:594:14)
E/flutter (14419): #11     _StreamImplEvents.handleNext  (dart:async/stream_impl.dart:710:11)
E/flutter (14419): #12     _PendingEvents.schedule.<anonymous closure>  (dart:async/stream_impl.dart:670:7)
E/flutter (14419): #13     _rootRun  (dart:async/zone.dart:1180:38)
E/flutter (14419): #14     _CustomZone.run  (dart:async/zone.dart:1077:19)
E/flutter (14419): #15     _CustomZone.runGuarded  (dart:async/zone.dart:979:7)
E/flutter (14419): #16     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1019:23)
E/flutter (14419): #17     _rootRun  (dart:async/zone.dart:1184:13)
E/flutter (14419): #18     _CustomZone.run  (dart:async/zone.dart:1077:19)
E/flutter (14419): #19     _CustomZone.runGuarded  (dart:async/zone.dart:979:7)
E/flutter (14419): #20     _CustomZone.bindCallbackGuarded.<anonymous closure>  (dart:async/zone.dart:1019:23)
E/flutter (14419): #21     _microtaskLoop  (dart:async/schedule_microtask.dart:43:21)
E/flutter (14419): #22     _startMicrotaskLoop  (dart:async/schedule_microtask.dart:52:5)
E/flutter (14419):
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 4 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 5 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 12 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 12 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 15 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 13 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 12 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 1 line
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 9 lines
W/System  (14419): A resource failed to call close.
W/System  (14419): A resource failed to call close.
I/chatty  (14419): uid=10383(com.example.music_taming) FinalizerDaemon identical 11 lines
W/System  (14419): A resource failed to call close.
```
When I've manually swiped resizedFile.lengthSync() < srcSize in _scaleImageFile with resizedFile.existsSync() && resizedFile.lengthSync() < srcSize first error disappeared but "A resource failed to call close" didn't go anywhere.

**Comment by humblerookie** [comment]
That's a good catch, assuming the compression fails, I'd like to fall back on the original file.

>but "A resource failed to call close"
Like I've said before this is a result of using the compression library,  as a matter of asserting the same you could just remove [this block](https://github.com/humblerookie/optimized_cached_image/commit/9e6752739263d7683005b0cd605d305048621ea2#diff-1a0cc0a6d95dd4b90ef0a33b24f26697R50) and verify those messages disappear.

**Comment by humblerookie** [comment]
This issue has been resolved now with[ release 0.1.14](https://pub.dev/packages/optimized_cached_image#-installing-tab-)
